isotovideo

```diff
@@ -256,6 +256,7 @@ checkout_git_repo_and_branch('NEEDLES_DIR');
 
 bmwqemu::ensure_valid_vars();
 
+$SIG{__WARN__} = sub { die "warning treated fatal (disable with 'NO_FATAL_WARNINGS'): @_" } unless $bmwqemu::vars{NO_FATAL_WARNINGS};
```

Comment:
It would be good to pass this as reason to openQA using `bmwqemu::serialize_state`.

I'd also like to note that this will not only affect the test code but also all code executed within isotovideo and all subsequently forked sub processes such as the command server. However, the commit message states that this change would only affect warnings within the test distribution code. So I would rather unset this after test modules have been loaded and explicitly enable it within the autotest process.

Reply:
do you think it would work if our die handler in general would do the state serialization?

Reply:
Now it actually already does that for the backend and isotovideo processes.

Reply:
so is your comment already resolved then?

Reply:
Not really. It looks like this would also affect subsequently started process and not only test distribution code. So the impact of this change might be quite big. So I would rather unset this after test modules have been loaded and explicitly enable it within the autotest process.

Reply:
hm, and if I strip the "within test distribution code" from the commit message to make it apply to all executed code, would it be fine? :)

Reply:
Maybe. If all our code was perfect and wouldn't produce any warnings it would definitely be fine. However, I'm not sure how "warning free" the code of os-autoinst (and of course any 3rdparty Perl module we use) in reality is. It might be that you provoke problems here which so far haven't been any. But if you want to generate some incompletes for your auto-review script to work with...

Reply:
https://lore.kernel.org/linux-block/CAHk-=wiFwUUEpCQ2FDWxBFb3XSxu=+yODHGfLPncvTQa9BeP9w@mail.gmail.com/t/ 

:-)

Reply:
Proving my point. We want to prevent any application to continue on any warning. Otherwise angry Linus will yell at us! And I guess nobody of us wants for that to happen ;)

Reply:
Wasn't he talking about warnings which can be detected at compile time? This type of warnings can be easily detected an addressed. We should have already a similar check.

This PR is affecting *runtime* warnings which are much harder to detect and fix in advance. Hence it is hard to say how much trouble we'll gain by merging this PR. In the best case it won't be notable change. I'd still consider it good if we find a few problems this way in our code although it means we're producing incomplete jobs until a fix is deployed. I'm just afraid that too many problems will come to the surface at once so we will have a hard time to fix these at once and explaining our users why there are now so many incomplete tests.

If you want to compare with the kernel: I suppose the kernel is actually very preservative when it comes to producing fatal errors (kernel panic) as it rarely makes sense to stop the entire system.

Reply:
Treating all runtime warnings as exceptions will definitely go wrong at some point. All it takes is a deprecation in one of the modules we depend on and nothing would work anymore.